test/functional/api/vim_spec.lua

```diff
@@ -1267,5 +1268,55 @@ describe('API', function()
       eq(expected, nvim("list_uis"))
     end)
   end)
-
+  describe('statusline', function()
+    it('last char shows (multibyte)', function()
+      local screen = Screen.new(20, 4)
+      screen:attach()
+      command('set laststatus=2')
+      command('set statusline=你好')
+      screen:expect([[
+      ^                    |
+      ~                   |
+      你好                |
+                          |
+      ]])
+    end)
+    it('last char shows', function()
+      local screen = Screen.new(20, 4)
+      screen:attach()
+      command('set laststatus=2')
+      command('set statusline=abc')
+      screen:expect([[
+      ^                    |
+      ~                   |
+      abc                 |
+                          |
+      ]])
+    end)
+    it('unicode control points', function()
+      local screen = Screen.new(20, 4)
+      screen:attach()
+      command('set laststatus=2')
+      feed(':set statusline=<C-v>u9f<CR>')
+      screen:expect([[
+      ^                    |
+      ~                   |
+      <9f>                |
+                          |
+      ]])
+    end)
+    it('MAX_MCO (6) unicode combination points', function()
+      local screen = Screen.new(24, 4)
+      screen:attach()
+      command('set laststatus=2')
+      command('set statusline=o̸⃯ᷰ⃐⃧⃝')
+      --o + U+1DF0 + U+20EF + U+0338 + U+20D0 + U+20E7 + UU+20DD
```

Comment:
UU+20DD -> U+20DD

Also same test with leading non-printable character followed by six combining characters?

Tests could be written a bit shorter if we use something like this
```lua
    before_each(function()
      --clear()
      screen = Screen.new(40, 4)
      screen:attach()
      command('set laststatus=2')
    end)

    after_each(function()
        screen:detach()
    end)

    ...

    it('last char shows', function()
      command('set statusline=abc')
      screen:expect([[
      ^                                        |
      ~                                       |
      abc                                     |
                                              |
      ]])
    end)

```

Reply:
Ok i've done that.